x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java

```diff
@@ -154,6 +154,8 @@ public void testFailsOnChecksumMismatch() {
         final RepositoryAnalyzeAction.Request request = new RepositoryAnalyzeAction.Request("test-repo");
         request.maxBlobSize(ByteSizeValue.ofBytes(10L));
         request.abortWritePermitted(false);
+        // Rare actions such as writeAndOverwrite and earlyReads can lead to CI instability
```

Comment:
Ah not quite, the overwrite happens in the "read phase", reads that are concurrent with the overwrite do not count as early:

https://github.com/elastic/elasticsearch/blob/3031595d6a234c31abe9f7da74a86a72d30d0652/x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java#L106-L117

Reply:
I could some help to understand the code. I don't quite get how Read phase starts before Overwrite. Is this the relevant code?

https://github.com/elastic/elasticsearch/blob/3031595d6a234c31abe9f7da74a86a72d30d0652/x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java#L282-L286

It seems to me that Ovewrite starts first with `writeRandomBlob` and the Read starts right before the last byte of the new content is written?

Reply:
That sounds about right, but this isn't an early read.

Reply:
OK I think I understand your point now. This regular read can also possibly return the old content because the overwrite is still ongoing and may not complete before the read returns. So yeah, this read also needs the lenient checksum comparison. I have adjusted the comment accordingly.

Reply:
Yep that's right LGTM